src/AppInstallerRepositoryCore/Rest/Schema/1_0/Json/SearchRequestSerializer.cpp

```diff
@@ -20,6 +20,48 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0::Json
         constexpr std::string_view MatchType = "MatchType"sv;
         constexpr std::string_view PackageMatchField = "PackageMatchField"sv;
         constexpr std::string_view FetchAllManifests = "FetchAllManifests"sv;
+
+        std::optional<std::string_view> ConvertPackageMatchFieldToString(AppInstaller::Repository::PackageMatchField field)
+        {
+            // Rest API schema supports only a subset of fields.
+            switch (field)

```

Comment:
@JohnMcPMS - Should the rest source also support NormalizedNameAndPublisher and any other match types apart from the ones below?
I am skipping the search request items that contain the match fields or match types not in the Rest API schema. Is this okay? If not, what is the expected behavior on the client side?

Reply:
Yes, it must support NormalizedNameAndPublisher in order to allow for local packages to be correlated with the remote ones.  I don't see why we wouldn't support all of the `PackageMatchField` and `MatchType` values here.

Why would those values not be in the schema?  The remote side is not obliged to implement them, but if they cannot even be sent it will create issues if/when we do want to use them.

In general, the behavior should be "fail as early as possible unless there is a reason not to".

Reply:
I have added the others as well.